text/trace/0220-messaging-semantic-conventions-span-structure.md

```diff
@@ -126,19 +126,22 @@ sending or publishing to an intermediary. This call or operation (and the
 related "Publish" span) can either refer to a single message or to a batch of
 multiple messages.
 
-It is recommended to create a "Create" span for every single message. "Create"
-spans can be created during the "Publish" operation as children of the
-"Publish" span. Alternatively, "Create" spans can be created independently of
-the "Publish" operation, e. g. in cases where messages are created before they
-are passed to a "Publish" operation. In this case, the "Publish" span should
-link to the "Create" spans.
-
-If a "Create" span exists for a message, its context must be injected into the
-message. If no "Create" span exists for a message, the context of the related
-"Publish" span must be injected into the message.
-
-"Create" spans must not be created for messages into which a context has
-already been injected.
+There are three different scenarios for injecting a creation context into a message:
+
+1. A user provides a custom creation context for a message, which doesn't refer
+   to any span described in this document. This provides flexibility to users
+   to model custom scenarios. In this case, no other spans besides the "Publish"
+   span should be created.
+2. If no custom creation context is provided for a message, it is recommended
+   to create a "Create" span for every single message and inject its context.
+   "Create" spans can be created during the "Publish" operation as children of the
+   "Publish" span.  Alternatively, "Create" spans can be created independently of
```

Comment:
Isn't this sentence:

> Alternatively, "Create" spans can be created independently of the "Publish" operation, e. g. in cases where messages are created before they    are passed to a "Publish" operation

Duplicated/say the same thing as in point 1. above? 

Because "created independently of the "Publish" operation" is essentially "user-land" code and that would fall under the "A user provides a custom creation context for a message" on point 1. I feel we can remove it entirely. 

Also, this part of "Publish" must link to creation is not really related to the "header" of this numbered list (here are three different scenarios for injecting a creation context into a message:). I see the value on talking about the links, so maybe make it a sub-item of 2. instead?

Reply:
We discussed this in the meeting (25.05) and others also agreed on the points above. @dpauls suggested also to remove the sentence "Anternatively..." and add something like this below points 1.

```
1. A user provides a custom creation context for a message, which doesn't refer
   to any span described in this document. This provides flexibility to users
   to model custom scenarios. In this case, no other spans besides the "Publish"
   span should be created, and the "Publish" span should link to this user provided
   context
```

Reply:
> Because "created independently of the "Publish" operation" is essentially "user-land" code and that would fall under the "A user provides a custom creation context for a message" on point 1. I feel we can remove it entirely.

Messaging SDKs can offer different API calls for creating and for publishing a message, that's the main intention of point (2) in the list above. E. g., this is from a Java Kafka SDK:

```java
final ProducerRecord<K, V> record = new ProducerRecord<>(topic, key, value);
Future<RecordMetadata> future = producer.send(record);
```

The constructor of `ProducerRecord` could be covered by a "Create" span (and its context being attached to the record), `producer.send` could be covered by a "Publish" span. This wouldn't really be "user-land" code, but instrumentation provided by an SDK.

Reply:
I see your point now. I think it's just a matter of how you look at it, since the create span was still created somewhere else and just passed to "send". I don't have any strong opinions on this, so feel free to close if you think what we have now is clear 👍